dataset.py

```diff
@@ -5,135 +5,212 @@
-  sample_data.drop('index', inplace=True, axis=1)
+def load_sample_data(reference_csv_filename: str) -> pd.DataFrame:
+    """This method loads reference sample data from a CSV, determines the unique
+  locations in that CSV and then calls the existing method
+  gen_tabular_dataset_with_coords to enrich those locations with features for
+  "rh", "temp", "vpd", and  "atmosphere_oxygen_ratio".
+  Finally, it joins the CSV provided d18O means values from the CSV with the
+  per-loc features"""
+
+    df = pd.read_csv(raster.get_sample_db_path(reference_csv_filename),
+                     encoding="ISO-8859-1", sep=',')
+    df = df[['Code', 'lat', 'long', 'd18O_cel']]
+    df = df[df['d18O_cel'].notna()]
+
+    grouped = df.groupby(['lat', 'long'])
+
+    # means is the reference sample calculated mean of d18O at each lat/lon
+    means = grouped.mean().reset_index()
+
+    # locations is now the list of unique lat and longs
+    locations = list(zip(means["long"], means["lat"]))
+
+    sample_data = gen_tabular_dataset_with_coords(monthly=False,
+                                                  samples_per_site=1, sample_site_coordinates=locations, sample_radius=0)
+    # drop the simulated cellulose_oxygen_ratio.
+    # TODO(https://github.com/tnc-br/ddf_common/issues/5), refactor the code to only do features.
+    sample_data = sample_data.drop('cellulose_oxygen_ratio', axis=1)
+
+    # Here we merge the features "rh", "temp", "vpd", and  "atmosphere_oxygen_ratio"
+    # with the means based on lat/long
+    sample_data = pd.merge(sample_data, means, how="inner",
+                           left_on=['lat', 'lon'], right_on=['lat', 'long'])
+    sample_data = sample_data.drop('long', axis=1).rename(
+        columns={'d18O_cel': 'cellulose_oxygen_ratio'}).reset_index()
+    sample_data.drop('index', inplace=True, axis=1)
+
+    return sample_data
 
+# If a reference_csv_filename is requested, we use that over any simulated data.
 
-  return sample_data
 
-# If a reference_csv_filename is requested, we use that over any simulated data.
 def aggregate_reference_data(reference_csv_filename: str) -> pd.DataFrame:
-  if reference_csv_filename:
-    return load_sample_data(reference_csv_filename)
-  # This is the historical simulated input that has 17 samples x 15 sites
-  return  gen_tabular_dataset(monthly=False, samples_per_site=17)
+    if reference_csv_filename:
+        return load_sample_data(reference_csv_filename)
+    # This is the historical simulated input that has 17 samples x 15 sites
+    return gen_tabular_dataset(monthly=False, samples_per_site=17)
 
-def partitioned_reference_data(reference_csv_filename: str) -> PartitionedDataset:
-  partition_data = partition(aggregate_reference_data(reference_csv_filename))
-  print_split(partition_data)
-  return partition_data
 
+def partitioned_reference_data(reference_csv_filename: str) -> PartitionedDataset:
+    partition_data = partition(
```

Comment:
dont you want to pass in the partition strategy here (and therefore needs to be a parameter in this method as well?